aiokafka/coordinator/consumer.py

```diff
@@ -0,0 +1,529 @@
+            auto_commit_interval_ms (int): milliseconds between automatic
+                offset commits, if enable_auto_commit is True. Default: 5000.
+            assignors (list): List of objects to use to distribute partition
+                ownership amongst consumer instances when group management is
+                used. Default: [RoundRobinPartitionAssignor]
+            heartbeat_interval_ms (int): The expected time in milliseconds
+                between heartbeats to the consumer coordinator when using
+                Kafka's group management feature. Heartbeats are used to ensure
+                that the consumer's session stays active and to facilitate
+                rebalancing when new consumers join or leave the group. The
+                value must be set lower than session_timeout_ms, but typically
+                should be set no higher than 1/3 of that value. It can be
+                adjusted even lower to control the expected time for normal
+                rebalances. Default: 3000
+            session_timeout_ms (int): The timeout used to detect failures when
+                using Kafka's group managementment facilities. Default: 30000
+            retry_backoff_ms (int): Milliseconds to backoff when retrying on
+                errors. Default: 100.
+        """
+        super(AIOConsumerCoordinator, self).__init__(
+            client, loop=loop, group_id=group_id,
+            session_timeout_ms=session_timeout_ms,
+            heartbeat_interval_ms=heartbeat_interval_ms,
+            retry_backoff_ms=retry_backoff_ms)
+
+        self._api_version = api_version
+        self._enable_auto_commit = enable_auto_commit
+        self._auto_commit_interval_ms = auto_commit_interval_ms
+        self._assignors = assignors
+        self._subscription = subscription
+        self._partitions_per_topic = {}
+        self._cluster = client.cluster
+        self._cluster.request_update()
+        self._cluster.add_listener(self._handle_metadata_update)
+        self._auto_commit_task = None
+
+        if self._api_version >= (0, 9) and self.group_id is not None:
+            assert self._assignors, 'Coordinator requires assignors'
+        if self._enable_auto_commit:
+            if self._api_version < (0, 8, 1):
+                log.warning(
+                    'Broker version (%s) does not support offset'
+                    ' commits; disabling auto-commit.', self._api_version)
+            elif self.group_id is None:
+                log.warning('group_id is None: disabling auto-commit.')
+            else:
+                interval = self._auto_commit_interval_ms / 1000.0
+                self._auto_commit_task = ensure_future(
+                    self.auto_commit_routine(interval), loop=loop)
+
+    @asyncio.coroutine
+    def close(self):
+        yield from self._maybe_auto_commit_offsets_sync()
+        if self._auto_commit_task:
+            self._auto_commit_task.cancel()
```

Comment:
We probably should use a different method here. How about we set a`closing` coroutine here and wait for autocommit task to finish. In the task replace all sleeps with `wait([self._closing])` calls. It will assure we will try at least 1-ce to perform autocommit before closing